src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.ValueGenerator/Helper/QueryHelper.cs

```diff
@@ -0,0 +1,172 @@
+{
+    /// <summary>
+    /// Helper class to easier work with queries
+    /// </summary>
+    internal static class QueryHelper
+    {
+        /// <summary>
+        /// a list of all value generator descriptions
+        /// </summary>
+        private static readonly CompositeFormat GeneratorDescriptionUuid = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uuid);
+        private static readonly CompositeFormat GeneratorDescriptionUuidv1 = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uuidv1);
+        private static readonly CompositeFormat GeneratorDescriptionUuidv3 = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uuidv3);
+        private static readonly CompositeFormat GeneratorDescriptionUuidv4 = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uuidv4);
+        private static readonly CompositeFormat GeneratorDescriptionUuidv5 = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uuidv5);
+        private static readonly CompositeFormat GeneratorDescriptionHash = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_hash);
+        private static readonly CompositeFormat GeneratorDescriptionBase64 = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_base64);
+        private static readonly CompositeFormat GeneratorDescriptionBase64d = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_base64d);
+        private static readonly CompositeFormat GeneratorDescriptionUrl = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_url);
+        private static readonly CompositeFormat GeneratorDescriptionUrld = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_urld);
+        private static readonly CompositeFormat GeneratorDescriptionEscData = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_esc_data);
+        private static readonly CompositeFormat GeneratorDescriptionUescData = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uesc_data);
+        private static readonly CompositeFormat GeneratorDescriptionEscHex = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_esc_hex);
+        private static readonly CompositeFormat GeneratorDescriptionUescHex = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_uesc_hex);
+        private static readonly CompositeFormat GeneratorDescriptionYourInput = System.Text.CompositeFormat.Parse(Properties.Resources.generator_description_your_input);
+        private static readonly CompositeFormat GeneratorExample = System.Text.CompositeFormat.Parse(Properties.Resources.generator_example);
+        private static readonly CompositeFormat Or = System.Text.CompositeFormat.Parse(Properties.Resources.or);
```

Comment:
Why do we need this? Can't we access directly.

Reply:
I'm just copying what already implemented here
These are some examples:
- https://github.com/microsoft/PowerToys/blob/3b7adbe6acc381d11d8e17866da4b79444679c08/src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.UnitConverter/Main.cs#L31
- https://github.com/microsoft/PowerToys/blob/3b7adbe6acc381d11d8e17866da4b79444679c08/src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.WebSearch/Main.cs#L42-L44
- https://github.com/microsoft/PowerToys/blob/3b7adbe6acc381d11d8e17866da4b79444679c08/src/modules/launcher/Plugins/Microsoft.Plugin.Folder/Sources/Result/EnvironmentVariableResult.cs#L16

Is there any examples where the resources can be accessed directly? Maybe I overlook it.

Reply:
Yes there is in Main.cs of the plugin: `public string Description => Resources.plugin_description;`
This requires the using `using Community.PowerToys.Run.Plugin.ValueGenerator.Properties;`

Reply:
Got it, thank you.